lib/rust/frp/src/io/timer/interval.rs

```diff
@@ -0,0 +1,97 @@
+//! FRP bindings for browser `setInterval` timer.
+
+use crate::prelude::*;
+pub use enso_web::traits::*;
+
+use crate as frp;
+
+use enso_web::window;
+use enso_web::Closure;
+
+
+// ================
+// === Interval ===
+// ================
+
+/// Closure type alias for use in `setInterval` call.
+type TimerClosure = Closure<dyn FnMut()>;
+
+/// Periodic timer.
+///
+/// The timer can be started or stopped at any time using `start` and `stop` inputs. After it is
+/// started, an `on_interval` event will be emitted periodically with period provided during
+/// start. The events will be emitted until the timer is stopped.
+///
+/// in `start`:        5ms-------------2ms---------------3ms------
+/// in `stop`:         -----------------------------x-----x-------
+/// out `on_interval`: -----x-----x-------x--x--x--x--------------
+///
+/// The timer is based on `setInterval` browser API. That means there is no guarantee about the
+/// exact time the events will be emitted. The true period between events will approach provided
+/// value over time.
+#[derive(Clone, CloneRef, Debug)]
+pub struct Interval {
+    /// Starts the timer with provided period value, specified in integer milliseconds. If the
+    /// timer was already started, it is restarted with new period value.
+    pub start:       frp::Any<i32>,
+    /// Stops the timer. No `on_interval` events will be emitted until it is started again.
+    pub stop:        frp::Any,
+    /// Triggered periodically after the timer is started.
+    pub on_interval: frp::Stream<()>,
+    raw_interval:    Rc<RawInterval>,
+}
+
+impl Interval {
+    /// Constructor. Timer is initially not started.
+    pub fn new(network: &frp::Network) -> Self {
+        frp::extend! { network
+            on_interval <- any_mut();
+        }
+
+        let closure: TimerClosure = Closure::new(f!(() on_interval.emit(())));
+        let raw_interval = Rc::new(RawInterval::new(closure));
+
+        frp::extend! { network
+            start <- any_mut::<i32>();
+            stop  <- any_mut::<()>();
+            eval  start((duration) raw_interval.start(*duration));
+            eval_ stop(raw_interval.stop());
+        }
+        let on_interval = on_interval.into();
+        Self { on_interval, start, stop, raw_interval }
+    }
+}
+
+#[derive(Debug)]
+struct RawInterval {
```

Comment:
section missing

Reply:
This is a private struct which is an implementation detail of `Interval`. I have intentionally put it in the same section. Do you think it should be in a separate one?

Reply:
Yup, we are putting such things in a separate section